### ISSUE TITLE
bitcoinj build fails with JDK 18

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/bitcoinj#2367]
The error I am seeing is `CertificateParsingException: No data available in passed DER encoded value`.

Here's the full stack-trace:

```
Testing with default toolchain
2022-03-31T01:21:30.2144600Z > Task :bitcoinj-core:test
2022-03-31T01:21:30.2145177Z Testing with default toolchain
2022-03-31T01:21:58.1146287Z 
2022-03-31T01:21:58.1147274Z org.bitcoinj.protocols.payments.PaymentProtocolTest > testSignAndVerifyValid FAILED
2022-03-31T01:21:58.1148473Z     org.bitcoinj.protocols.payments.PaymentProtocolException$PkiVerificationException: java.security.cert.CertificateParsingException: No data available in passed DER encoded value.
2022-03-31T01:21:58.1149733Z         at app//org.bitcoinj.protocols.payments.PaymentProtocol.verifyPaymentRequestPki(PaymentProtocol.java:228)
2022-03-31T01:21:58.1150758Z         at app//org.bitcoinj.protocols.payments.PaymentProtocolTest.testSignAndVerifyValid(PaymentProtocolTest.java:81)
2022-03-31T01:21:58.1151265Z 
2022-03-31T01:21:58.1151555Z         Caused by:
2022-03-31T01:21:58.1152086Z         java.security.cert.CertificateParsingException: No data available in passed DER encoded value.
2022-03-31T01:21:58.1152799Z             at java.base/sun.security.x509.X509CertImpl.getExtensionIfParseable(X509CertImpl.java:1477)
2022-03-31T01:21:58.1153505Z             at java.base/sun.security.x509.X509CertImpl.getSubjectAlternativeNames(X509CertImpl.java:1639)
2022-03-31T01:21:58.1154207Z             at org.bitcoinj.crypto.X509Utils.getDisplayNameFromCertificate(X509Utils.java:69)
2022-03-31T01:21:58.1154990Z             at org.bitcoinj.protocols.payments.PaymentProtocol.verifyPaymentRequestPki(PaymentProtocol.java:216)
2022-03-31T01:21:58.1155584Z             ... 1 more
2022-03-31T01:22:34.3144179Z 
2022-03-31T01:22:34.3144633Z 30861 tests completed, 1 failed, 8155 skipped
2022-03-31T01:22:35.1176189Z 
2022-03-31T01:22:35.1177404Z > Task :bitcoinj-core:test FAILED
```

You can see the failure on Github CI  [here](https://github.com/bitcoinj/bitcoinj/runs/5763931400?check_suite_focus=true) via PR #2366

**Comment by schildbach** [comment]
So the change is it doesn't return `null` and more if the "subjectAlternativeNames" don't exist? Rather, it throws the above exception?

**Comment by msgilligan** [comment]
Fixed by #2368.
